lm_eval/models/huggingface.py

```diff
@@ -499,31 +518,49 @@ def _model_generate(
 
 
 class MultiTokenEOSCriteria(transformers.StoppingCriteria):
-    """Criteria to stop on the specified multi-token sequence."""
+    """
+    Criteria to stop on the specified multi-token sequence.
+    """
```

Comment:
This still fails the unit tests for causal language models [here](https://github.com/bigscience-workshop/lm-evaluation-harness/blob/4c9a9caf3453a93cdb582315eb3c93c3c6d2eaef/tests/test_models_huggingface.py#L30).  You can test it out locally using: 

`pytest -vv -k "test_stop_sequences"`

For example with gpt2, using a custom stop sequence, "great",  for gpt2 (tokenized to `18223`) the model seems to continue generating after encountering the string. 

This goes back to our discussion of whether we should allow for strings that get tokenized differently in subtle ways that are effectively  "undetectable" for stopping. E.g.
`gpt2_tokenizer.encode(" great") -> 1049 != 18223 <- gpt2_tokenizer.encode("great")`